kentikapi/internal/api_resources/sites_test.go

```diff
@@ -0,0 +1,209 @@
+	assert.Equal("mysite", site.SiteName)
+	assert.Equal(1.0, *site.Latitude)
+	assert.Equal(2.0, *site.Longitude)
+	assert.Equal(models.ID(74333), site.CompanyID)
+}
+
+func TestCreateSite(t *testing.T) {
+	// arrange
+	createResponsePayload := `
+	{
+		"site": {
+			"id": 42,
+			"site_name": "apitest-site-1",
+			"lat": 54.349276,
+			"lon": 18.659577,
+			"company_id": "3250"
+		}     
+	}`
+	transport := &api_connection.StubTransport{ResponseBody: createResponsePayload}
+	sitesAPI := api_resources.NewSitesAPI(transport)
+
+	// act
+	site := models.NewSite("apitest-site-1")
+	models.SetOptional(&site.Longitude, 18.659577)
+	models.SetOptional(&site.Latitude, 54.349276)
+	created, err := sitesAPI.Create(nil, *site)
+
+	// assert request properly formed
+	assert := assert.New(t)
+	require := require.New(t)
+
+	require.NoError(err)
+	payload := utils.NewJSONPayloadInspector(t, transport.RequestBody)
+	require.NotNil(payload.Get("site"))
+	assert.Equal("apitest-site-1", payload.String("site/site_name"))
+	assert.Equal(18.659577, payload.Float("site/lon"))
+	assert.Equal(54.349276, payload.Float("site/lat"))
+
+	// and response properly parsed
+	assert.Equal(models.ID(42), created.ID)
+	assert.Equal("apitest-site-1", created.SiteName)
+	assert.Equal(54.349276, *created.Latitude)
+	assert.Equal(18.659577, *created.Longitude)
+	assert.Equal(models.ID(3250), created.CompanyID)
+}
+
+func TestUpdatetSite(t *testing.T) {
+	// arrange
+	updateResponsePayload := `
+	{
+		"site": {
+			"id": "42",
+			"site_name": "new-site",
+			"lat": -15.0,
+			"lon": -45.0,
+			"company_id": "3250"
+		}
+	}`
+	transport := &api_connection.StubTransport{ResponseBody: updateResponsePayload}
+	sitesAPI := api_resources.NewSitesAPI(transport)
+
+	// act
+	siteID := models.ID(42)
+	site := models.Site{ID: siteID, SiteName: "new-site"}
+	models.SetOptional(&site.Longitude, -45.0)
+	updated, err := sitesAPI.Update(nil, site)
+
+	// assert request properly formed
+	assert := assert.New(t)
+	require := require.New(t)
+
+	require.NoError(err)
+	payload := utils.NewJSONPayloadInspector(t, transport.RequestBody)
+	require.NotNil(payload.Get("site"))
+	assert.Nil(payload.Get("site/lat"))
+	assert.Equal("new-site", payload.String("site/site_name"))
+	assert.Equal(-45.0, payload.Float("site/lon"))
+
+	// # and response properly parsed
+	assert.Equal(models.ID(42), updated.ID)
+	assert.Equal("new-site", updated.SiteName)
+	assert.Equal(-15.0, *updated.Latitude)
+	assert.Equal(-45.0, *updated.Longitude)
+	assert.Equal(models.ID(3250), updated.CompanyID)
+}
+
+func TestDeleteSite(t *testing.T) {
+	// arrange
+	deleteResponsePayload := "" // deleting site responds with empty body
+	transport := &api_connection.StubTransport{ResponseBody: deleteResponsePayload}
+	sitesAPI := api_resources.NewSitesAPI(transport)
+
+	// act
+	siteID := models.ID(42)
+	err := sitesAPI.Delete(nil, siteID)
+
+	// assert
+	assert := assert.New(t)
+	require := require.New(t)
+	require.NoError(err)
```

Comment:
Nit: please use `require.NoError(t, err)` form so that we have 2 lines instead of 4

Reply:
Resolved